src/Soil/Soil.jl

```diff
@@ -72,15 +71,18 @@ import ClimaLSM:
     auxiliary_vars,
     name,
     prognostic_types,
-    auxiliary_types
+    auxiliary_types,
+    AbstractSource,
+    AbstractBC,
+    source!,
+    boundary_flux
 export RichardsModel,
     RichardsParameters,
     EnergyHydrology,
     EnergyHydrologyParameters,
-    boundary_flux,
-    AbstractBC,
     FluxBC,
```

Comment:
Should `FluxBC` and `StateBC` exist in SharedUtilities/boundary_conditions.jl? If they're only used by soil models I think it makes sense to keep them here, but if they can be used more generally it might be good to keep all the boundary conditions objects in one place.

Reply:
I think I am on the fence with this! It would great to have them stored and used generally, but each model will have a different method for how to turn a state into a flux. so we cant have a single StateBC type for all models.
but maybe we can somehow get around this because these will all be of the form: factor_at_center*(BC - state)/dz?

Reply:
holding off on this: will address longer term https://github.com/CliMA/ClimaLSM.jl/issues/116